<a href="https://colab.research.google.com/github/allanacardoso/introducao_analise_de_dados/blob/master/Desafio_Aula3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### **Importando a biblioteca e lendo o arquivo**

In [ ]:
import pandas as pd

In [ ]:
dados = pd.read_excel ("AdventureWorks.xlsx")
dados


Data Venda Data Envio  ID Loja  ...            Marca   Classe    Cor
0   2008-05-09 2008-05-29      199  ...  Adventure Works  Regular    Red
1   2008-05-12 2008-05-17      306  ...  Adventure Works  Regular    Red
2   2008-05-14 2008-05-20      306  ...  Adventure Works  Regular    Red
3   2008-05-21 2008-05-27      306  ...  Adventure Works  Regular    Red
4   2008-06-20 2008-06-27      306  ...  Adventure Works  Regular    Red
..         ...        ...      ...  ...              ...      ...    ...
899 2008-03-22 2008-04-04      306  ...         Fabrikam  Regular  Black
900 2008-10-04 2008-10-15      306  ...         Fabrikam  Regular  Black
901 2008-12-11 2008-12-15      306  ...         Fabrikam  Regular  Black
902 2008-12-17 2008-12-21      306  ...         Fabrikam  Regular  Black
903 2008-12-21 2008-12-25      306  ...         Fabrikam  Regular  Black

[904 rows x 16 columns]

### **Qual a quantidade de linhas e colunas que a base possui?**


In [ ]:
dados.count

<bound method DataFrame.count of     Data Venda Data Envio  ID Loja  ...            Marca   Classe    Cor
0   2008-05-09 2008-05-29      199  ...  Adventure Works  Regular    Red
1   2008-05-12 2008-05-17      306  ...  Adventure Works  Regular    Red
2   2008-05-14 2008-05-20      306  ...  Adventure Works  Regular    Red
3   2008-05-21 2008-05-27      306  ...  Adventure Works  Regular    Red
4   2008-06-20 2008-06-27      306  ...  Adventure Works  Regular    Red
..         ...        ...      ...  ...              ...      ...    ...
899 2008-03-22 2008-04-04      306  ...         Fabrikam  Regular  Black
900 2008-10-04 2008-10-15      306  ...         Fabrikam  Regular  Black
901 2008-12-11 2008-12-15      306  ...         Fabrikam  Regular  Black
902 2008-12-17 2008-12-21      306  ...         Fabrikam  Regular  Black
903 2008-12-21 2008-12-25      306  ...         Fabrikam  Regular  Black

[904 rows x 16 columns]>

**904 rows x 16 columns**

### **Exibir as 5 últimas linhas da base de dados**


In [ ]:
dados.tail

<bound method NDFrame.tail of     Data Venda Data Envio  ID Loja  ...            Marca   Classe    Cor
0   2008-05-09 2008-05-29      199  ...  Adventure Works  Regular    Red
1   2008-05-12 2008-05-17      306  ...  Adventure Works  Regular    Red
2   2008-05-14 2008-05-20      306  ...  Adventure Works  Regular    Red
3   2008-05-21 2008-05-27      306  ...  Adventure Works  Regular    Red
4   2008-06-20 2008-06-27      306  ...  Adventure Works  Regular    Red
..         ...        ...      ...  ...              ...      ...    ...
899 2008-03-22 2008-04-04      306  ...         Fabrikam  Regular  Black
900 2008-10-04 2008-10-15      306  ...         Fabrikam  Regular  Black
901 2008-12-11 2008-12-15      306  ...         Fabrikam  Regular  Black
902 2008-12-17 2008-12-21      306  ...         Fabrikam  Regular  Black
903 2008-12-21 2008-12-25      306  ...         Fabrikam  Regular  Black

[904 rows x 16 columns]>

### **Quais são os tipos de dados de cada coluna**





In [ ]:
dados.dtypes



Data Venda        datetime64[ns]
Data Envio        datetime64[ns]
ID Loja                    int64
ID Produto                 int64
ID Cliente                 int64
No. Venda                 object
Custo Unitário           float64
Preço Unitário           float64
Quantidade                 int64
Valor Desconto           float64
Valor Venda              float64
Produto                   object
Fabricante                object
Marca                     object
Classe                    object
Cor                       object
dtype: object

### **Qual a Receita Total?**


In [ ]:
dados["Receita"] = dados["Valor Venda"].mul(dados["Quantidade"])
dados.head()

Data Venda Data Envio  ID Loja  ...   Classe  Cor  Receita
0 2008-05-09 2008-05-29      199  ...  Regular  Red  27288.0
1 2008-05-12 2008-05-17      306  ...  Regular  Red  27288.0
2 2008-05-14 2008-05-20      306  ...  Regular  Red  27288.0
3 2008-05-21 2008-05-27      306  ...  Regular  Red  27288.0
4 2008-06-20 2008-06-27      306  ...  Regular  Red  27288.0

[5 rows x 17 columns]

### **Qual o custo Total?**


In [ ]:
dados["Receita-Custo Unitário"] = dados["Receita"] - dados["Custo Unitário"]

In [ ]:
dados.head()

Data Venda Data Envio  ID Loja  ...  Cor  Receita Receita-Custo Unitário
0 2008-05-09 2008-05-29      199  ...  Red  27288.0               26939.42
1 2008-05-12 2008-05-17      306  ...  Red  27288.0               26939.42
2 2008-05-14 2008-05-20      306  ...  Red  27288.0               26939.42
3 2008-05-21 2008-05-27      306  ...  Red  27288.0               26939.42
4 2008-06-20 2008-06-27      306  ...  Red  27288.0               26939.42

[5 rows x 18 columns]

### **Qual o Lucro Total?**


In [ ]:
dados.groupby("ID Loja")["Receita"].sum()

ID Loja
199    9.253737e+07
306    5.986742e+07
307    1.235462e+08
Name: Receita, dtype: float64

### **Qual foi o Lucro por ano?**


In [ ]:
dados.groupby(dados['Data Venda'].dt.year)['Receita-Custo Unitário'].sum()


Data Venda
2008    6.649066e+07
2009    2.092144e+08
Name: Receita-Custo Unitário, dtype: float64

### **Qual o total de produtos vendidos?**


In [ ]:
dados.groupby(dados["Quantidade"]).sum()

ID Loja  ID Produto  ...       Receita  Receita-Custo Unitário
Quantidade                       ...                                      
2               199         384  ...  2.880400e+03            2.531820e+03
4              9792       29137  ...  3.734272e+05            3.626894e+05
5             17563       59378  ...  1.280930e+06            1.259149e+06
6             67936      241622  ...  7.985308e+06            7.900144e+06
7              6517       19405  ...  9.000869e+05            8.923307e+05
8             19830       59915  ...  3.622311e+06            3.598408e+06
9             10150       32991  ...  2.284334e+06            2.271564e+06
10             7433       25492  ...  2.021365e+06            2.012688e+06
11             4875       16654  ...  1.712852e+06            1.706815e+06
12            20084       73083  ...  9.701896e+06            9.676849e+06
13             2437        9955  ...  1.330870e+06            1.327893e+06
14             4875       18981  ...  2.984284e+06            2.978332e+06
15             4875       15853  ...  3.103366e+06            3.097306e+06
16             6990       24522  ...  5.233769e+06            5.225034e+06
17             1916        6357  ...  1.714550e+06            1.711839e+06
18             4762       16573  ...  5.153674e+06            5.147655e+06
19              307         384  ...  2.736380e+05            2.732894e+05
20              903        3948  ...  1.280480e+06            1.279139e+06
21              812        2747  ...  1.069204e+06            1.068201e+06
22             1317        5050  ...  1.986404e+06            1.984744e+06
23              505        1562  ...  8.518699e+05            8.511906e+05
24             2128        8782  ...  4.199777e+06            4.197125e+06
25              506        2381  ...  1.020719e+06            1.020063e+06
26              199        1226  ...  6.077916e+05            6.074606e+05
28              199        1226  ...  7.048944e+05            7.045634e+05
30              506        1521  ...  1.461870e+06            1.461197e+06
32              505        1587  ...  1.423770e+06            1.423096e+06
37              306        2496  ...  1.230868e+04            1.230359e+04
44              199        2496  ...  1.547251e+04            1.546742e+04
48              199        2497  ...  2.301696e+04            2.301187e+04
74              199        2496  ...  4.376419e+04            4.375910e+04
96             1224        9985  ...  2.946171e+05            2.945967e+05
108            1224        9987  ...  4.194841e+05            4.194637e+05
120            4472       39943  ...  2.301696e+06            2.301615e+06
144            3542       34954  ...  2.610123e+06            2.610052e+06
156            3650       34950  ...  2.820153e+06            2.820082e+06
192            3672       29956  ...  3.535405e+06            3.535344e+06
216            2238       19973  ...  3.216045e+06            3.216004e+06
240            5472       52423  ...  1.208390e+07            1.208380e+07
264             199        2496  ...  5.570104e+05            5.570053e+05
288            4966       47436  ...  1.438468e+07            1.438458e+07
312            4662       44938  ...  1.448975e+07            1.448966e+07
384            1224        9986  ...  4.713873e+06            4.713853e+06
428             199        2497  ...  1.830008e+06            1.830003e+06
432            1427       12482  ...  7.923588e+06            7.923563e+06
468             199        2496  ...  1.750440e+06            1.750435e+06
480            1319       12482  ...  1.150848e+07            1.150845e+07
576            2130       19973  ...  2.439429e+07            2.439425e+07
624            1120        9987  ...  1.303105e+07            1.303103e+07
720             506        4993  ...  1.035763e+07            1.035762e+07
864             307        2496  ...  6.711746e+06            6.711740e+06
936             199        2496  ... 